### PR TITLE
finished custom client metadata support

### PR DESCRIPTION
**Pull request by sethwright**

### CODE REVIEW

File: test/authlete/model/test_client.rb [hunk 1]
```diff
@@ -115,6 +115,7 @@ class ClientTest < Minitest::Test
   ATTRIBUTE_KEY                              = '<attribute0-key>'
   ATTRIBUTE_VALUE                            = '<attribute0-value>'
   ATTRIBUTES                                 = [ Authlete::Model::Pair.new(key: ATTRIBUTE_KEY, value: ATTRIBUTE_VALUE) ]
+  CUSTOM_METADATA                            = 'JSON string'
```
Comment (hidebike712) [inline test/authlete/model/test_client.rb:118]: Can you repalce the `'JSON String'` with `'<custom-matadata>'` ?

File: test/authlete/model/test_service.rb [hunk 2]
```diff
@@ -142,6 +142,7 @@ class ServiceTest < Minitest::Test
   ATTRIBUTE_KEY                                 = '<attribute0-key>'
   ATTRIBUTE_VALUE                               = '<attribute0-value>'
   ATTRIBUTES                                    = [ Authlete::Model::Pair.new(key: ATTRIBUTE_KEY, value: ATTRIBUTE_VALUE) ]
+  SUPPORTED_CUSTOM_CLIENT_METADATA              = 'JSON string'
```
Comment (hidebike712) [inline test/authlete/model/test_service.rb:145]: Can you replace the `'JSON String'` with `[ '<supported-custom-client-metadata0>', '<supported-custom-client-metadata1>' ]` ?

NOTE: `supportedCustomClientMetadata` is a string array object, not a string object.

File: test/authlete/model/test_client.rb (hunk 1 above)
Comment (sethwright) [reply test/authlete/model/test_client.rb]: I have made the change.

File: test/authlete/model/test_service.rb (hunk 2 above)
Comment (sethwright) [reply test/authlete/model/test_service.rb]: I have made the change.

**Pull request merged**